bin/knc.c

```diff
@@ -795,7 +794,10 @@ move_local_to_network_buffer(work_t *work)
 		/* EOF */
 		return 0;
 	} else if (work->network_buffer.in_len < 0) {
-		if (errno == ECONNRESET)
+		const int errnum = errno;
```

Comment:
We don't need this `errnum`.  We only need this when we'll call things like `close(2)` that could stomp on the current value of `errno`, and when we do my preference is to call that `save_errno`.

Reply:
As you said, it is not needed in these cases to preserve errno.  It is an idiom I have seen used and used myself for a long time to store errno in errnum for reuse, rather than repeating a function call to obtain the (often) thread-local errno value.

Would you prefer that I repeat the use of errno rather than storing in errnum?

Reply:
changed to repeat use of errno.